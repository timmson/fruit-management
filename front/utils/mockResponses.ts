export function okAndJson(json: any): Promise<Response> {
    return Promise.resolve({
        arrayBuffer(): Promise<ArrayBuffer> {
            return Promise.resolve(undefined);
        },
        blob(): Promise<Blob> {
            return Promise.resolve(undefined);
        },
        body: undefined,
        bodyUsed: false,
        clone(): Response {
            return undefined;
        },
        formData(): Promise<FormData> {
            return Promise.resolve(undefined);
        },
        headers: undefined,
        json(): Promise<any> {
            return Promise.resolve(json);
        },
        ok: true,
        redirected: false,
        status: 200,
        statusText: "OK",
        text(): Promise<string> {
            return Promise.resolve("");
        },
        type: undefined,
        url: ""
    })
}