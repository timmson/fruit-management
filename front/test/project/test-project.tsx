import React from "react"
import renderer from "react-test-renderer"
import Context from "../../src/context"
import {okAndJson} from "../../utils/mockResponses"
import Project from "../../src/project/project"

global.fetch = jest.fn(() => okAndJson({}))

describe("Project should", () => {

	test("return project form", () => {
		const context = []

		const component = renderer.create(
			<Context.Provider value={context}>
				<Project/>
			</Context.Provider>
		)

		expect(component.toJSON()).toMatchSnapshot()

		component.unmount()
	})

})