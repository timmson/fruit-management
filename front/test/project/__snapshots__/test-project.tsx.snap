// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should return project form 1`] = `
[
  <h2>
    Проекты
  </h2>,
  <table
    className="basic"
  >
    <thead>
      <tr>
        <th>
          Проект
        </th>
        <th>
          График
        </th>
      </tr>
    </thead>
    <tbody />
  </table>,
]
`;
